api/src/main/java/org/pmiops/workbench/notebooks/NotebooksServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package org.pmiops.workbench.notebooks;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+import java.util.List;
+import javax.inject.Provider;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.notebooks.api.ClusterApi;
+import org.pmiops.workbench.notebooks.api.NotebooksApi;
+import org.pmiops.workbench.notebooks.api.TestApi;
+import org.pmiops.workbench.notebooks.model.Cluster;
+import org.pmiops.workbench.notebooks.model.ClusterRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class NotebooksServiceImpl implements NotebooksService {
+
+  private final Provider<ClusterApi> clusterApiProvider;
+  private final Provider<NotebooksApi> notebooksApiProvider;
+  private final Provider<TestApi> testApiProvider;
+  private final Provider<User> userProvider;
+
+  @Autowired
+  public NotebooksServiceImpl(Provider<ClusterApi> clusterApiProvider,
+      Provider<NotebooksApi> notebooksApiProvider,
+      Provider<TestApi> testApiProvider,
+      Provider<User> userProvider) {
+    this.clusterApiProvider = clusterApiProvider;
+    this.notebooksApiProvider = notebooksApiProvider;
+    this.testApiProvider = testApiProvider;
+    this.userProvider = userProvider;
+  }
+
+  @Override
+  public Cluster createCluster(String googleProject, String clusterName, ClusterRequest clusterRequest)
+      throws ApiException {
+    ClusterApi clusterApi = clusterApiProvider.get();
+    try {
+      clusterName = clusterName + this.userProvider.get().getUserId();
```

Comment:
I'm wondering if this logic (constructing the cluster name) belongs outside the cluster.

Either way, we should have a helper method so we don't repeat the logic in create / delete / get.

Reply:
I'll make it a helper. Could you clarify what you mean by putting the logic inside the cluster? Edit the constructer function?

Reply:
Er sorry, I meant in the controller rather than internal to the service.

(For Firecloud, I've been trying to put business logic in the controller / outside the service classes, and make the service classes as dumb as possible... that way we can write unit tests of the controller logic and exercise all the business logic with mocks.)